File: perl/src/tests/t_decrement_slave_count.pl
#!/usr/bin/perl
#
# $Id:$
#
# perl test driver

use Atrshmlog;

$result = Atrshmlog::attach();

Atrshmlog::sleep_nanos(1000000000);

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++$akcount;

    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "decrement slave count : $akcount : \n";

$s = Atrshmlog::create_slave();

Atrshmlog::sleep_nanos(1000000000);

print "decrement slave count : $s : \n";

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++$akcount;

    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "decrement slave count : $akcount : \n";

$r = Atrshmlog::decrement_slave_count();

print "decrement slave count : $r : \n";

while ($r >  0) {
    $r = Atrshmlog::decrement_slave_count();
    print "decrement slave count : $r : \n";
}

$r = Atrshmlog::decrement_slave_count();

print "decrement slave count : $r : \n";

print " \n";

exit (0);

# end of main

File: python/src/tests/t_decrement_slave_count.py
#!/usr/bin/python3
#
# $Id:$
#
# We test a bit of the atrshmlog here.
#
# This is for the first starter, so only the basic things.

import sys
import atrshmlog

r = atrshmlog.attach()

atrshmlog.sleep_nanos(1000000000)

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while slave != 0:
    akcount = akcount + 1
    slave = atrshmlog.get_next_slave_local(slave)


print('decrement slave count : ' + str(akcount) + ' : ')

s = atrshmlog.create_slave()

atrshmlog.sleep_nanos(1000000000)

print('decrement slave count : ' + str(s) + ' : ')

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while slave != 0:
    akcount = akcount + 1
    slave = atrshmlog.get_next_slave_local(slave)


print('decrement slave count : ' + str(akcount) + ' : ')

r = atrshmlog.decrement_slave_count()

print('decrement slave count : ' + str(r) + ' : ')


while r > 0:
    r = atrshmlog.decrement_slave_count()
    print('decrement slave count : ' + str(r) + ' : ')

r = atrshmlog.decrement_slave_count()

print('decrement slave count : ' + str(r) + ' : ')

print (' ')

exit(0);

# end of test

